### PR TITLE
OSX backend calls OnResized to keep WindowState inSync.

### PR DESCRIPTION
**Pull request by danwalmsley**
## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

File: native/Avalonia.Native/src/OSX/window.mm [hunk 1]
```diff
@@ -1360,6 +1371,16 @@ - (void)windowDidDeminiaturize:(NSNotification *)notification
     }
 }
 
+- (void)windowDidResize:(NSNotification *)notification
+{
+    auto parent = dynamic_cast<IWindowStateChanged*>(_parent.operator->());
```
Comment (kekekeks) [inline native/Avalonia.Native/src/OSX/window.mm]: Why is this needed?

Comment (danwalmsley) [reply native/Avalonia.Native/src/OSX/window.mm]: removed cast, and interface method. not required.
